feat: Add Chat Completion API support to SharedPrefixDataGenerator

bongwoobak opened the pull request:
## Summary
Extended SharedPrefixDataGenerator to support Chat Completion API alongside the existing Completion API.

## Changes
- Add Chat Completion API support to SharedPrefixDataGenerator
- Shared prefix --> system message, question --> user message
- Backward compatible with existing Completion API

## Modified Files
- `inference_perf/datagen/shared_prefix_datagen.py`

bongwoobak committed:
add chat completion in shared prefix datagen

SachinVarghese commented:
/assign @Bslabe123 
Make sure compatibility with multi-turn

bongwoobak committed:
feat: Add Chat API + Multi-turn support

- Implement UserSessionChatAPIData for Chat API with multi-turn chat
- Update SharedPrefixDataGenerator to support Chat API + Multi-turn combination
- Fix prompt selection to use different questions for each turn
- Initialize user session context as list for Chat API compatibility

bongwoobak commented:
Sorry for the delay. I've updated the code to support multi-turn chat properly.

jjk-g commented:
should fix #303

Bslabe123 commented on inference_perf/apis/user_session.py:
```diff
@@ -7,22 +7,24 @@
 class LocalUserSession:
     user_session_id: str
-    context: str
+    user_session_id: str
```
Duplicate `user_session_id`?

bongwoobak committed:
feat: Fix multi-turn chat group_id tracking and add tests

jjk-g commented:
https://github.com/kubernetes-sigs/inference-perf/pull/331 seems to also address shuffle.

Let's revert this PR to the chat completion API support, and follow up the multi-turn shuffle PR with the tests you added in the later commit.

Thanks!

bongwoobak committed:
refactor: Remove multi-turn Chat API support per reviewer request

bongwoobak committed:
test: Add unit tests for SharedPrefixDataGenerator
Also revert shuffle synchronization to match origin/main - only shuffle
prompts list. Shuffle synchronization for prompt_pairs and user_sessions
will be handled separately in PR #331.

Bslabe123 commented:
Thanks as well for adding tests! very much appreciated

bongwoobak commented:
@Bslabe123 Can we progress with this?

hhk7734 committed:
test: Set integer seed in create_data_config mock

Mocked shared_prefix.seed was a Mock, causing
np.random.default_rng(...) to raise TypeError in SharedPrefixDataGenerator
tests. Pin it to an int so the RNG initializes.

Signed-off-by: Hyeonki Hong <hhk7734@gmail.com>
